perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java

```diff
@@ -40,55 +44,88 @@
 	private final static Logger log = LoggerFactory.getLogger(Vsup.class);
 
 	@Override
-	public List<ApplicationFormItemData> createApplication(PerunSession session, Application application, List<ApplicationFormItemData> data) throws PerunException {
+	public void canBeApproved(PerunSession session, Application app) throws PerunException {
 
-		if (application.getUser() == null) {
+		List<ApplicationFormItemData> data = registrar.getApplicationDataById(session, app.getId());
+
+		if (app.getUser() == null) {
+
+			for (ApplicationFormItemData item : data) {
+				if (item.getFormItem() != null &&
+					Objects.equals(AttributesManager.NS_USER_ATTR_DEF + ":birthNumber", item.getFormItem().getPerunDestinationAttribute())) {
+
+					// if application contains birth number, try to map to existing user
+					String rc = item.getValue();
+					if (rc != null && !rc.isEmpty()) {
+
+						try {
+							User user = ((PerunBl) session.getPerun()).getUsersManagerBl().getUserByExtSourceNameAndExtLogin(session, "RC", rc);
+							throw new CantBeApprovedException("Application has the same birth number " + rc + " as user " + user.getDisplayName() + " with id " + user.getId() +
+								" that is already in Perun and thus would be merged with him.", null, null, null, true);
+						} catch (CantBeApprovedException ex) {
+							throw ex;
+						} catch (Exception ex) {
+							log.warn("Couldn't find or set user to application {} by RC: {}", app, ex);
```

Comment:
Are you sure this should react to any exception? Because if there will be for example "InternalErrorException", this will say that the application can be approved if I understand it right.
@zlamalp should it work this way?

Reply:
Ok, I've change the catched exception only to those that are thrown by `getUserByExtSourceNameAndExtLogin()`.